### PR TITLE
Add HTTPS/domain support via `url` crate, bump edition/version, add CI and tests

### PR DESCRIPTION
**Pull request by code-tc**
### Motivation

- Support HTTPS URLs, domain names (including IDNA) and offload host parsing/normalization to a dedicated URL parser instead of custom ad-hoc parsing.
- Ensure correctness across feature combinations by adding a CI job matrix that runs the test suite for common feature sets.
- Modernize the crate metadata and dependencies (edition bump and dependency version updates) to align with maintained crates.

### Description

- Added a GitHub Actions workflow at `.github/workflows/ci.yml` to run `cargo test` across a feature matrix including `--no-default-features`, default, `--all-features`, and individual features `hash`, `stream`, `serialize`, and `uuid`.
- Reworked host parsing to delegate parsing and IDNA/domain normalization to the `url` crate by introducing `use url::Url;` and parsing `Host` from a `Url::parse` result, and added rejection checks for userinfo, query, fragment, and unexpected path components.
- Introduced `Protocol::HTTPS` and `Address::Domain(String)` variants and updated ordering, `as_ip`, `is_localhost`, and `GetSize` implementations accordingly.
- Bumped crate `version` to `0.5.0`, `edition` to `2024`, updated dependency versions (including `destream = "0.10"`, `hr-id = "0.7"`, added `url = "2.5"`) and added `serde_json` in `[dev-dependencies]`.
- Added unit tests exercising host/link parsing, IDNA normalization and rejection cases in `src/lib.rs`, serde roundtrip tests in `src/serial.rs`, and destream trait presence tests in `src/stream.rs`, and a minor import reordering in `src/path.rs`.

### Testing

- Added unit tests in `src/lib.rs`, `src/serial.rs`, and `src/stream.rs` and ran the test suite with `cargo test`, and all tests passed locally.
- Added a CI workflow to run `cargo test ${{ matrix.args }}` for multiple feature combinations so the same unit tests will be executed automatically on push and pull requests.

------
[Codex Task](https://chatgpt.com/codex/cloud/tasks/task_b_69fb5876fd60832189f0fbf6a823d32f)

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -710,6 +956,26 @@ mod tests {
...
+    }
+
+    #[test]
+    fn parse_link_rejects_unsupported_protocol() {
+        let err = "ftp://example.com/a"
+            .parse::<Link>()
+            .expect_err("unsupported protocol should fail");
+        assert!(err.to_string().contains("invalid protocol"));
+    }
+
```
Comment (aliaksandr-vasilenka-vention) [inline src/lib.rs:989]: It seems we are missing edge case test here. if we try parse link from "https://example.com". Result link via to_string() will produce "https://example.com/" with / at the end. Because url crate sets the PathBuf to root /. Is it expected behaviour?

**Review by aliaksandr-vasilenka-vention — APPROVED** [review]
